runtime/lua/vim/lsp/handlers.lua

```diff
@@ -565,6 +565,35 @@ M[ms.callHierarchy_incomingCalls] = make_call_hierarchy_handler('from')
 --- @see # https://microsoft.github.io/language-server-protocol/specifications/specification-current/#callHierarchy_outgoingCalls
 M[ms.callHierarchy_outgoingCalls] = make_call_hierarchy_handler('to')
 
+--- @private
+---
+--- Displays type hierarchy in the quickfix window.
+local function make_type_hierarchy_handler()
+  --- @param result lsp.TypeHierarchyItem[]
+  return function(_, result)
+    if not result then
+      return
+    end
+    local items = {}
+    for _, type_hierarchy_item in pairs(result) do
+      table.insert(items, {
+        filename = assert(vim.uri_to_fname(type_hierarchy_item.uri)),
+        text = type_hierarchy_item.name,
+        lnum = type_hierarchy_item.range.start.line + 1,
+        col = type_hierarchy_item.range.start.character + 1,
```

Comment:
This needs to consider the offset encoding of the client/server, and convert the (likey UTF-16) offset into byte offsets. You can search for `offset_encoding` to see how this translation is done in other places.